File: frontend/src/app/models/category.ts
import {Page} from './page';
import {ProductDto} from './product';

export interface Category {
  id: number;
  name: string;
  description: string | null;
  nameEn: string | null;
  descriptionEn: string | null;
  slug: string;
}

export interface CategoryProducts {
  category: Category;
  productsPage: Page<ProductDto>;
}

File: backend/src/main/java/pl/dudios/shop/common/model/Category.java
package pl.dudios.shop.common.model;

import lombok.Getter;
import lombok.Setter;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

@Entity
@Getter
@Setter
@Table(name = "categories")
public class Category {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String description;
    // Wersja angielska (opcjonalna) — null/pusty = frontend pokazuje polską.
    private String nameEn;
    private String descriptionEn;
    private String slug;

}
